### PR TITLE
Reworks and rebalances Geis, merges conversion sigils

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
tweak: Geis no longer converts. Instead, it traps the target in bindings that they must struggle to break free from, and can be dragged around freely. They will break after 25 seconds of resisting or when struck with the invoker's slab.
balance: Geis now takes one second to invoke (down from two) and has two invocations (down from four.)
balance: The sigil of submission is now permanent, but cannot destroy mindshield implants. The sigil of accession has been removed.
/:cl:

### This is a change I'd like to see testmerged a bit before it's added into the game.

I'm calling this the conversion patch.

Geis is no longer used for conversion. Instead, it functions as a brief stun/mute, as well as a way to restrain any non-Servants for nefarious deeds. Attacking a non-Servant with the charged slab will apply geis bindings to them, muting them for four ticks and stunning them for two seconds, as well as making them unable to use any items.

The geis bindings have 25 health and can thus be attacked in normal ways to break the victim free. Additionally, the victim can begin resisting, which will damage the bindings for 1 health per second. This resisting can only be interrupted if the victim goes unconscious, dies, or becomes a Servant.

They also lose health at a rate of 1 per second when not being dragged by a Servant.

_Note:_ While the target is bound with Geis, the invoker's slab is unusable, and only becomes usable once the bindings break. Because of this, striking the bindings with the slab now destroys them instantly, which will allow Servants choice in how they proceed.

The primary conversion tool is now the sigil of submission, which has been made a Script tier scripture. It now costs 1 geis capacitor per use, and remains permanently after conversions. However, it still cannot break mindshield implants. Because the sigil of accession has been removed, this means that **mindshielded targets can no longer be converted without surgery, body swapping, or other methods being done first to remove the implant.**

I made these changes because the conversion aspect of clockcult isn't very clear. Early on, you're told that Geis is the best way to convert people, and use that. Then, after you get used to using it, it suddenly becomes useless, forcing you to rely on other tools and sigils, which leads to confusion and people not understanding what to do.

We took a lesson from bloodcult. They have stunpapers - a clear stunning tool - and conversion runes, a clear conversion tool. Geis now fills the role of the stunning tool used to capture and subdue unruly targets and potential converts, and the sigil of submission now fills the role of the clear and obvious method of conversion.

### tl;dr I made some handy dandy flowcharts, this is how stuff looks now:
[Old Conversion](http://i.imgur.com/1chOQAL.png) | [New Conversion](http://i.imgur.com/tjdJVo3.png)

@ChangelingRain

### CODE REVIEW

**Comment by Repukan** [comment]
>25 second binds

😢😢😢😢😢

**Comment by Xhuis** [comment]
They're also guaranteed to be escapable unless you're critted, unlike bloodcult's classic "stunpaper -> cablecuffs", which can't be escaped without help, if they slip up and you get away from them, or if they decide your fate somehow.

These are easily escaped on a timer unless you're knocked unconscious, meaning that it's a race against time for the Servant. It makes conversion simpler and less confusing throughout the round while still not trivializing it.

**Comment by Ergovisavi** [comment]
>Can't break them off a friend with your bare hands

Oversight or intended? I'm not particularly convinced an inability to assist a friend due to not carrying a fire extinguisher on your person at all times is great.

**Comment by Xhuis** [comment]
Rescuing someone using your bare hands is as simple as grabbing the bindings. They fall apart pretty rapidly when not being dragged by a Servant, so if you can snag your buddy away from the bad guy and escape, you're home free.

File: code/game/gamemodes/clock_cult/clock_helpers/slab_abilities.dm [hunk 1]
```diff
@@ -69,12 +69,13 @@
...
-				conversion.slab = slab
-				conversion.invoker = ranged_ability_user
-				conversion.target = target
-				conversion.run_scripture()
+				if(L.buckled)
+					L.buckled.unbuckle_mob(L, TRUE)
+				var/obj/structure/destructible/clockwork/geis_binding/binding = new(get_turf(target))
+				binding.setDir(target.dir)
+				binding.buckle_mob(L, TRUE)
+				binding.slab = slab
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_helpers/slab_abilities.dm]: make the slab an argument for the binding instead

File: code/game/gamemodes/clock_cult/clock_structures/geis_binding.dm [hunk 2]
```diff
@@ -16,15 +17,58 @@
...
-	var/resisting = FALSE
-	var/can_resist = FALSE
+	var/mob/living/resisting = FALSE
 	var/mob_layer = MOB_LAYER
+	var/obj/item/clockwork/slab/slab
+
+/obj/structure/destructible/clockwork/geis_binding/Initialize()
+	. = ..()
+	START_PROCESSING(SSprocessing, src)
+	addtimer(CALLBACK(src, .proc/make_slab_busy), 1) //This is needed due to how Initialize() handles
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_structures/geis_binding.dm]: and then you don't have to do this

File: code/game/gamemodes/clock_cult/clock_structures/geis_binding.dm [hunk 3]
```diff
@@ -106,19 +149,15 @@
...
-	else
-		return ..()
+	if(resisting)
+		to_chat(user, "<span class='warning'>You're already trying to break free!</span>")
+		return
+	if(is_servant_of_ratvar(user))
+		take_damage(obj_integrity) //freedom!
+		return
+	resisting = user
+	user.visible_message("<span class='warning'>[user] starts struggling against [src]...</span>", "<span class='userdanger'>You start breaking out of [src]...</span>")
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_structures/geis_binding.dm]: this should be automatic on binding someone rather than when they resist; it should be a flat duration no matter what the bound target is doing, at least unless they die

File: code/game/gamemodes/clock_cult/clock_scriptures/scripture_drivers.dm [hunk 4]
```diff
@@ -97,82 +97,46 @@
...
+	channel_time = 60
+	consumed_components = list(GEIS_CAPACITOR = 1)
+	whispered = TRUE
+	object_path = /obj/effect/clockwork/sigil/submission
+	creator_message = "<span class='brass'>A luminous sigil appears below you. The next non-servant to cross it will be enslaved after a brief time if they do not move.</span>"
+	usage_tip = "This is the primary method of converting heretics into Servants. Geis is an excellent tool for restraining potential converts."
+	tier = SCRIPTURE_SCRIPT
+	one_per_tile = TRUE
+	primary_component = GEIS_CAPACITOR
+	sort_priority = 5
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_scriptures/scripture_drivers.dm]: I would greatly recommend cutting taunting tirade at the same time and making this 6

Comment (Xhuis) [inline code/game/gamemodes/clock_cult/clock_scriptures/scripture_drivers.dm]: I actually missed that variable, thank you for the heads-up.
While I want to remove taunting tirade, it's probably for another PR, unfortunately.

**Comment by Ratvar** [comment]
Can you use second slab to draw convert sygils on a go? Geis target,pull,swap hands, draw brassrune of submission with second slab,

**Comment by Xhuis** [comment]
I'm not sure. I don't think so, but I'll have to check. I can't test right now (on mobile) so it may be a while.

**Comment by KingPhilipIII** [comment]
I love the flow charts. "Apply murder."

But yea. I'm interested to see how this goes. Clockcult's conversion system was always a bit convoluted in how it was supposed to be done, even once you figured out how it worked, so this definitely simplifies things.

Although. Since I dislike change for the most part, a small part of me is resistant to radically altering one of the core concepts of this antagonist.

File: code/game/gamemodes/clock_cult/clock_structures/geis_binding.dm (hunk 3 above)
Comment (Xhuis) [inline code/game/gamemodes/clock_cult/clock_structures/geis_binding.dm]: My main concern is that people won't understand that they're being broken out and start saying "hey this is bullshit I can't break free", so I'm kind of thinking that it might be worth keeping in for that

File: code/game/gamemodes/clock_cult/clock_structures/geis_binding.dm [hunk 5]
```diff
@@ -16,15 +17,60 @@
...
+	. = ..()
+	START_PROCESSING(SSprocessing, src)
+
+/obj/structure/destructible/clockwork/geis_binding/Destroy()
+	STOP_PROCESSING(SSprocessing, src)
+	slab.busy = null
+	slab.icon_state = initial(slab.icon_state)
+	return ..()
+
+/obj/structure/destructible/clockwork/geis_binding/proc/assign_slab(obj/item/clockwork/slab/the_slab) //feed me energy
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_structures/geis_binding.dm]: this should be in Initialize, with the sleep()ed part under an addtimer() with a time of 1

**Comment by DoctoreMad** [comment]
Does the geis thing have enough health to warrant giving it a special proc when hit by the chaplain's weapon?

**Comment by Xhuis** [comment]
It has 25 health, meaning that two hits from a void weapon will destroy it. I don't think it warrants anything, no, but that's just my own opinion.

**Comment by GuyonBroadway** [comment]
I dislike the changes to how the runes work because it treads a bit too much on the toes of blood cult.

File: code/game/gamemodes/clock_cult/clock_structures/geis_binding.dm [hunk 6]
```diff
@@ -16,15 +17,61 @@
...
+/obj/structure/destructible/clockwork/geis_binding/Destroy()
+	STOP_PROCESSING(SSprocessing, src)
+	slab.busy = null
+	slab.icon_state = initial(slab.icon_state)
+	return ..()
+
+/obj/structure/destructible/clockwork/geis_binding/proc/assign_slab(obj/item/clockwork/slab/the_slab) //retuuuuuuuuurn the_slab
+	if(!the_slab)
+		return
+	slab = the_slab
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_structures/geis_binding.dm]: this should be in initialize, under the if(slab)

File: code/game/gamemodes/clock_cult/clock_structures/geis_binding.dm [hunk 7]
```diff
@@ -16,15 +17,61 @@
...
+		addtimer(CALLBACK(src, .proc/assign_slab, slab), 1)
+
+/obj/structure/destructible/clockwork/geis_binding/Destroy()
+	STOP_PROCESSING(SSprocessing, src)
+	slab.busy = null
+	slab.icon_state = initial(slab.icon_state)
+	return ..()
+
+/obj/structure/destructible/clockwork/geis_binding/proc/assign_slab(obj/item/clockwork/slab/the_slab) //retuuuuuuuuurn the_slab
+	if(!the_slab)
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_structures/geis_binding.dm]: if(QDELETED(the_slab))

File: code/game/gamemodes/clock_cult/clock_structures/geis_binding.dm [hunk 8]
```diff
@@ -16,15 +17,61 @@
...
+	if(slab == I)
+		user.visible_message("<span class='warning'>[user] dispels [src]!</span>", "<span class='danger'>You dispel the bindings!</span>")
+		take_damage(obj_integrity)
+		return
+	. = ..()
+
+/obj/structure/destructible/clockwork/geis_binding/process()
+	if(!pulledby || !is_servant_of_ratvar(pulledby))
+		take_damage(1) //Quickly decays when not pulled by a servant
+	if(!resisting)
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_structures/geis_binding.dm]: if(QDELETED(resisting))

File: code/game/gamemodes/clock_cult/clock_structures/geis_binding.dm (hunk 7 above)
Comment (Xhuis) [inline code/game/gamemodes/clock_cult/clock_structures/geis_binding.dm]: Does that also cover `!the_slab`?

**Comment by GuyonBroadway** [comment]
Also you might wanna rename Geis then, because Geis is an old irish myth about a curse that compels you to do something... DnD then made it a spell that does the same.

Change its name to something like "Bind" or "Restrain" or "Apprehend"

**Comment by Xhuis** [comment]
I was considering the name Geis Shackles. In keeping with the theme of "free driver scripture = first word of primary component" it might be breaking theme to rename it. If I do rename anything, it'll be "Geis Shackles" and the submission sigil to "Geis Sigil', so keep that theme.

File: code/game/gamemodes/clock_cult/clock_structures/geis_binding.dm [hunk 9]
```diff
@@ -22,22 +22,22 @@
...
-	if(slab)
-		addtimer(CALLBACK(src, .proc/assign_slab, slab), 1)
+	if(!QDELETED(the_slab) && istype(the_slab))
+		slab = the_slab
+		addtimer(CALLBACK(src, .proc/assign_slab), 1)
 
 /obj/structure/destructible/clockwork/geis_binding/Destroy()
 	STOP_PROCESSING(SSprocessing, src)
 	slab.busy = null
 	slab.icon_state = initial(slab.icon_state)
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_structures/geis_binding.dm]: slab = null after this line

**Comment by Bombpants** [comment]
I liked how mobile clock cultists could be with the original geis. Once they start building they are easy to spot as they lack the hiding rune ability blood has.

**Comment by BlueNothing** [comment]
I think I rather prefer the previous version, with the former style of Geis and the Sigils of Submission and Accession as presently. Stun-sigil + Submission is a neat mid-game trap, Geis is a solid way to mid-game if used properly, and Sigil of Accession makes the third step neatly. Collectively, it makes a neat, if somewhat unintuitive, progression, but as a team antag, the cultists should be teaching each other.

**Comment by Xhuis** [comment]
Thank you for the feedback! I'm aiming to run a few more rounds with it today, and make the decision from there if I should close or tweak this. While I agree that Geis is more mobile and that that's a good thing, its current version is very clunky, and I'd like to do something with it. Maybe I could, like, make the restraints function, but hitting them with the slab starts the convert? vOv

**Comment by drfino** [comment]
I think this makes clock cult too similar to blood cult. Permanent conversion rune on the floor, temporary (very long) stun that is mobile. This change makes the play styles nearly identical for the first ~10 minutes.

**Comment by ChangelingRain** [comment]
Have Geis change the mouse cursor to a cancel button after binding, accordingly preventing you from doing stuff as long as the binding exists.
Let it cancel from any range+when clicking your slab+maybe if you drop your slab.

This solves having a second slab to make a sigil with while still letting it be effective.

Also compared to stunpapers this is much more obvious and accordingly limited, since(with my above change) you won't be able to hit someone with it and stun/cuff/whatever them on your own, though you could still convert.

**Comment by KingPhilipIII** [comment]
So what you're  saying, and your argument for why this isn't the same thing as bloodcult, @ChangelingRain  is that we have a shittier version of the stunpaper in that it's more obvious, less useful, and otherwise functions the same for the ultimate mechanics of conversion.

I'll be honest, I'm still not convinced.

**Comment by KingPhilipIII** [comment]
More or less unrelated, but I just realized clock cultists missed out on an easy way to cheese seccies. Charge up to a seccie. Geis them. While they're held immobile until they resist out of it have a buddy wail on them. when they break free if they don't just go into crit they're severely injured.

**Comment by DoctoreMad** [comment]
Wait how do you convert AIs now?

**Comment by Xhuis** [comment]
You don't, because I'm a butt and forgot to include a failsafe for it. Once I get some time to do some more work on this, I'll do something about that.

**Comment by ChangelingRain** [comment]
Hitting an AI with geis already force-buckles it, letting you drag it around. Should work.

**Comment by Xhuis** [comment]
I'll have to test that, but it should work, yeah.

**Comment by Xhuis** [comment]
Making Geis prevent you from doing _any_ actions sounds incredibly restrictive to me, and I think it would just end up making people bitter. If at all possible I'm going to try to make it _more_ simple to convert, not less so.

**Comment by ChangelingRain** [comment]
I mean, if they can do anything, it's a 25 second stun, long enough to remove radio, run off, draw a sigil, and convert them.
Or to beat them into crit with a crowbar and have time to spare.

Accordingly, either make it restricted or make it a no additional bullshit normal stun with the same or slightly less power than a stunpaper.

**Comment by Xhuis** [comment]
I'm leaning towards making it a more generic, targeted stun. I may also add restraining etc. capabilities into the slab ala abductor baton, but that's something I'm still tossed up on.

**Comment by ChangelingRain** [comment]
I would greatly prefer it being restricted, since that has less issues. Any stun longer than 8 seconds is basically lethal.

**Comment by Xhuis** [comment]
I'd prefer having it do something other than a stun, too. Playtesting is showing, though, that while the new Geis is _more_ intuitive, it still has its problems, and most people didn't realize they had to put down the sigil until I expressly told them. Restricting all actions with it would only amplify that.

It's a problem that doesn't have an easy solution. Moving their reliance toward sigils of submission makes them very similar to bloodcult, but less intuitive. With the two options we have now it's either "worse version of bloodcult" or "unintuitive mechanics, which clockcult already suffers from."

I'm gonna take a few hours to see if I can brainstorm any solutions besides those two that would still realistically work. If I have to restrict what Geis lets you do, I will, but only if there's no other solution. If you can think of any ideas, I'd appreciate any insight.

**Comment by ChangelingRain** [comment]
I feel like it's only unintuitive to them because it was recently changed, isn't yet on the wiki, and isn't actually merged yet. Compared to how it is currently, it's much more intuitive.

Also, when you make it restrict, make it have a different mouse cursor so they know they still have an override on.

**Comment by DoctoreMad** [comment]
Will the sigil of conversion become a default ability(IE: Automatically put in the skill hotbar) like Geis?

**Comment by Xhuis** [comment]
I don't see why not.

**Comment by king5327** [comment]
I was also concerned with how conversion in the earlygame works, but my proposition was different:
Geis and SigConvert stay the same, Accession is demoted from an implant-breaker to a permanent SigConvert (still breaks when trying to convert implanted).

The difference however, is that whenever a conversion happens, it leaves behind a conversion marker naming the newly converted. I also think that implanting clock cultists should fail, unlike blood which (if memory serves and there haven't been updates since) can get an implant while remaining loyal to Nar-Sie.

Just my two bits.

**Comment by Xhuis** [comment]
Just a heads-up to y'all: I'm getting overwhelmed with the amount of stuff I'm doing at once so I'm gonna put this PR on "potentially closed" for now. This one in particular wouldn't be coming back due to the freeze and I'd likely incorporate any of the feedback I got into the rework.

**Comment by Xhuis** [comment]
After a brief discussion, I've decided that the current state of Geis isn't important enough to justify this bypassing the freeze. Any major changes will either come after the freeze by @ChangelingRain (or me, if I'm able to) or with the rework.
